libraries/AP_Camera/AP_RunCam.h

```diff
@@ -0,0 +1,319 @@
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  implementation of RunCam camera protocols 
+
+  With thanks to betaflight for a great reference
+  implementation. Several of the functions below are based on
+  betaflight equivalent functions
+ */
+#pragma once
+
+#include <AP_Param/AP_Param.h>
+#include <RC_Channel/RC_Channel.h>
+#include <AP_RCMapper/AP_RCMapper.h>
+#include <AP_Arming/AP_Arming.h>
+
+// the protocol for Runcam Device definition
+#define RCDEVICE_PROTOCOL_HEADER 0xCC
+
+#define RCDEVICE_PROTOCOL_MAX_PACKET_SIZE 64
+#define RCDEVICE_PROTOCOL_MAX_DATA_SIZE 62
+
+#define RCDEVICE_MAX_WAITING_RESPONSES 1
+#define RCDEVICE_NUM_EXPECTED_RESPONSES 4
+#define RCDEVICE_OSD_MENU_DEPTH 2
+
+/// @class	AP_RunCam
+/// @brief	Object managing a RunCam device
+class AP_RunCam
+{
+public:
+    AP_RunCam(RCMapper* rcmapper)
+    {
+        AP_Param::setup_object_defaults(this, var_info);
+        _rcmapper = rcmapper;
+        _singleton = this;
+        _video_recording = true;
+        _is_ready = false;
+    }
+
+    // do not allow copies
+    AP_RunCam(const AP_RunCam &other) = delete;
+    AP_RunCam &operator=(const AP_RunCam &) = delete;
+
+    // get singleton instance
+    static AP_RunCam *get_singleton()
+    {
+        return _singleton;
+    }
+
+    // feature flag - uint16_t
+    enum RCFeatures
+    {
+        RCDEVICE_PROTOCOL_FEATURE_SIMULATE_POWER_BUTTON = (1 << 0),
+        RCDEVICE_PROTOCOL_FEATURE_SIMULATE_WIFI_BUTTON = (1 << 1), // WiFi/Mode button
+        RCDEVICE_PROTOCOL_FEATURE_CHANGE_MODE = (1 << 2),
+        RCDEVICE_PROTOCOL_FEATURE_SIMULATE_5_KEY_OSD_CABLE = (1 << 3),
+        RCDEVICE_PROTOCOL_FEATURE_DEVICE_SETTINGS_ACCESS = (1 << 4),
+        RCDEVICE_PROTOCOL_FEATURE_DISPLAY_PORT = (1 << 5),
+        RCDEVICE_PROTOCOL_FEATURE_START_RECORDING = (1 << 6),
+        RCDEVICE_PROTOCOL_FEATURE_STOP_RECORDING = (1 << 7)
+    };
+
+    // operation of camera button simulation
+    enum RCCameraControlOperation
+    {
+        RCDEVICE_PROTOCOL_CAM_CTRL_SIMULATE_WIFI_BTN = 0x00, // WiFi/Mode button
+        RCDEVICE_PROTOCOL_CAM_CTRL_SIMULATE_POWER_BTN = 0x01,
+        RCDEVICE_PROTOCOL_CAM_CTRL_CHANGE_MODE = 0x02,
+        RCDEVICE_PROTOCOL_CAM_CTRL_START_RECORDING = 0x03,
+        RCDEVICE_PROTOCOL_CAM_CTRL_STOP_RECORDING = 0x04,
+        RCDEVICE_PROTOCOL_CAM_CTRL_UNKNOWN_CAMERA_OPERATION = 0xFF
+    };
+
+    // camera control commands
+    enum RCCameraCommands
+    {
+        RCDEVICE_PROTOCOL_COMMAND_GET_DEVICE_INFO = 0x00,
```

Comment:
```suggestion
        GET_DEVICE_INFO = 0x00,
```

... and more enum-classification comments assumed from here on :-)